scripts/linux/cross-platform-rust-build.sh

```diff
@@ -0,0 +1,278 @@
+#!/bin/bash
+
+###############################################################################
+# Print usage information pertaining to this script and exit
+###############################################################################
+usage()
+{
+    echo "$SCRIPT_NAME [options]"
+    echo "This script is used for building a few rust components (i.e. broker, edge hub watchdog)."
+    echo "The logic here is similar to the edgelet packaging."
+    echo ""
+    echo "options"
+    echo " --os                      Desired os for build"
+    echo " --arch                    Desired arch for build"
+    echo " --build-path              Desired arch for build"
+    echo " -h, --help                Print this help and exit."
+    exit 1;
+}
+
+###############################################################################
+# Obtain and validate the options supported by this script
+###############################################################################
+process_args()
+{
+    save_next_arg=0
+    for arg in "$@"
+    do
+        if [ $save_next_arg -eq 1 ]; then
+            PACKAGE_OS="$arg"
+            save_next_arg=0
+        elif [ $save_next_arg -eq 2 ]; then
+            PACKAGE_ARCH="$arg"
+            save_next_arg=0
+        elif [ $save_next_arg -eq 3 ]; then
+            BUILD_PATH="$arg"
+            save_next_arg=0
+        else
+            case "$arg" in
+                "--os" ) save_next_arg=1;;
+                "--arch" ) save_next_arg=2;;
+                "--build-path" ) save_next_arg=3;;
+                "-h" | "--help" ) usage;;
+                * ) usage;;
+            esac
+        fi
+    done
+}
+
+process_args "$@"
+
+[[ -z "$PACKAGE_OS" ]] && { print_error 'OS is a required parameter'; exit 1; }
+[[ -z "$PACKAGE_ARCH" ]] && { print_error 'Arch is a required parameter'; exit 1; }
+[[ -z "$BUILD_PATH" ]] && { print_error 'Build path is a required parameter'; exit 1; }
+
+set -e
+
+# Get directory of running script
+DIR="$(cd "$(dirname "$0")" && pwd)"
+
+BUILD_REPOSITORY_LOCALPATH="$(realpath "${BUILD_REPOSITORY_LOCALPATH:-$DIR/../..}")"
+
+DOCKER_VOLUME_MOUNTS=''
+
+case "$PACKAGE_OS" in
+    'ubuntu18.04')
+        DOCKER_IMAGE='ubuntu:18.04'
+        ;;
+esac
+
+if [ -z "$DOCKER_IMAGE" ]; then
+    echo "Unrecognized target [$PACKAGE_OS.$PACKAGE_ARCH]" >&2
+    exit 1
+fi
+
+case "$PACKAGE_ARCH" in
+    'amd64')
+        RUST_TARGET='x86_64-unknown-linux-musl'
+        ;;
+
+    'arm32v7')
+        RUST_TARGET='armv7-unknown-linux-gnueabihf'
+        ;;
+
+    'aarch64')
+        RUST_TARGET='aarch64-unknown-linux-gnu'
+        ;;
+esac
+
+if [ -n "$RUST_TARGET" ]; then
+    RUST_TARGET_COMMAND="rustup target add $RUST_TARGET &&"
+fi
+
+
+case "$PACKAGE_OS.$PACKAGE_ARCH" in
+    ubuntu18.04.amd64)
+        # The below SETUP was copied from https://github.com/emk/rust-musl-builder/blob/master/Dockerfile.
```

Comment:
I think we should mention this in THIRDPARTYNOTICE file, since that repo is under MIT license.

Reply:
I added it. Do you know why though we have three different THIRDPARTYNOTICES files? I feel it would be better to just have one?